### ISSUE TITLE
Orgchart drag and drop node from a parent node to chart canvas

### ISSUE THREAD

**Issue by daekadog** [BALKANGraph/OrgChartJS#567]
Hi,

Is it possible to drag and drop a node to the chart canvas? In other words I want to move a node from a parent node to a non-parent (i.e. take it to the root level)

It seems the “drop” event does not capture this and will only work when dropping onto another node

Is there a javascript workaround for this that can help me

**Comment by plamen-peshev** [comment]
Get latest version

```
        var chart = new OrgChart(document.getElementById("tree"), {
            nodeBinding: {
                field_0: "name"
            },
            enableDragDrop: true
        });

        chart.on('drop', function (sender, draggedNodeId, droppedNodeId) {
            if (!droppedNodeId){
                sender.config.roots = [];
                for(var i = 0; i < sender.roots.length; i++){
                    sender.config.roots.push(sender.roots[i].id);
                }
                sender.config.roots.push(draggedNodeId);
                sender.draw();
            }
        });

        chart.load([
            { id: 1, name: "Denny Curtis" },
            { id: 2, pid: 1, name: "Ashley Barnett" },
            { id: 3, pid: 1, name: "Caden Ellison" },
            { id: 4, pid: 3, name: "Elliot Patel" },
            { id: 5, pid: 2, name: "Lynn Hussain" },
            { id: 6, pid: 3, name: "Tanner May" },
            { id: 7, pid: 3, name: "Fran Parsons" },
            { id: 8, pid: 4, name: "Fran Parsons" },
            { id: 9, pid: 8, name: "Fran Parsons" },
            { id: 10, pid: 9, name: "Fran Parsons" },
            { id: 11, pid: 10, name: "Fran Parsons" },
            { id: 12, pid: 12, name: "The Buggy One" }
        ]);
```

**Comment by daekadog** [comment]
Hi,

 

Confirmed that  having made the changes as described

 

Unfortunately it seems the chart.on('click') does not get triggered any more and instead the chart.on('drop') does

 

Look forward to fix

 

Regards

 

Pat Williams

CEO at Cornstalk Software Pty Ltd

TKO Business Modeller Software

1300 498 180 or (international) +61 419 337 287

 <https://tkodocs.com/> TKO Web Site

**Comment by ZornitsaSerbezova** [comment]
Could you give us a demo where chart.on("click") is not triggered?

**Comment by plamen-peshev** [comment]
We managed to replicate the issue, will fix it soon

**Comment by plamen-peshev** [comment]
The issue is addressed in version 7.8.06
